### PR TITLE
Add custom error class

### LINKED ISSUE #71: Write custom error classes

### PR DESCRIPTION
**Pull request by giovannt0**
Closes #71 

Introduces new error classes:

- CredentialsConfigurationError
- InvalidFileFormatError
- InvalidNameError
- ResourceNotFoundError
- ResourceCreationError
- ResourceDeletionError

Tests updated accordingly.

Signed-off-by: Theo Giovanna <gtheo@google.com>

### CODE REVIEW

File: libcloudforensics/providers/gcp/forensics.py [hunk 1]
```diff
@@ -88,28 +90,23 @@ def CreateDiskCopy(
...
-        'Try running:\n  $ gcloud auth application-default login')
-    raise RuntimeError(error_msg)
+  except (RefreshError, DefaultCredentialsError) as exception:
+    raise errors.CredentialsConfigurationError(
+        'Something is wrong with your Application Default Credentials. Try '
+        'running: $ gcloud auth application-default login: {0!s}'.format(
+            exception), __name__)
   except HttpError as exception:
     if exception.resp.status == 403:
       raise RuntimeError(
```
Comment (rgayon) [inline libcloudforensics/providers/gcp/forensics.py:100]: Would this be  a CredentialsConfigurationError ?

File: libcloudforensics/providers/gcp/forensics.py [hunk 2]
```diff
@@ -88,28 +90,23 @@ def CreateDiskCopy(
...
-      raise RuntimeError(
+      raise errors.ResourceNotFoundError(
           'GCP resource not found. Maybe a typo in the project / instance / '
-          'disk name?')
+          'disk name?', __name__)
     raise RuntimeError(exception)
   except RuntimeError as exception:
-    error_msg = 'Cannot copy disk "{0:s}": {1!s}'.format(disk_name, exception)
-    raise RuntimeError(error_msg)
+    raise RuntimeError(
```
Comment (rgayon) [inline libcloudforensics/providers/gcp/forensics.py:108]: What about ResourceCreationError?

File: libcloudforensics/providers/gcp/internal/common.py [hunk 3]
```diff
@@ -266,12 +267,10 @@ def ExecuteRequest(client: 'googleapiclient.discovery.Resource',
...
     except (RefreshError, DefaultCredentialsError) as exception:
-      error_msg = (
-          '{0:s}\n'
-          'Something is wrong with your Application Default '
-          'Credentials. Try running: '
-          '$ gcloud auth application-default login'.format(str(exception)))
-      raise RuntimeError(error_msg)
+      raise errors.CredentialsConfigurationError(
+          'Something is wrong with your Application Default Credentials. Try '
+          'running: $ gcloud auth application-default login: {0!s}'.format(
```
Comment (rgayon) [inline libcloudforensics/providers/gcp/internal/common.py:272]: I'd have the exception message before the gclout auth command, I believe it'd be easier to read

File: libcloudforensics/providers/aws/forensics.py [hunk 4]
```diff
@@ -17,7 +17,7 @@
 
 from libcloudforensics.providers.aws.internal.common import UBUNTU_1804_FILTER
 from libcloudforensics.providers.aws.internal import account
-from libcloudforensics import logging_utils
+from libcloudforensics import logging_utils, errors
```
Comment (rgayon) [inline libcloudforensics/providers/aws/forensics.py:20]: Does our styleguide allow multiple import?

Comment (giovannt0) [reply libcloudforensics/providers/aws/forensics.py:20]: I think we have this kind of multiple imports a bit everywhere. I opt to leave it as is for now and opened #243 to keep track.

File: libcloudforensics/providers/gcp/internal/common.py (hunk 3 above)
Comment (giovannt0) [reply libcloudforensics/providers/gcp/internal/common.py:272]: I think we have this kind of multiple imports a bit everywhere. I opt to leave it as is for now and opened

File: libcloudforensics/providers/gcp/forensics.py (hunk 2 above)
Comment (giovannt0) [reply libcloudforensics/providers/gcp/forensics.py:108]: Changed

File: libcloudforensics/providers/gcp/forensics.py (hunk 1 above)
Comment (giovannt0) [reply libcloudforensics/providers/gcp/forensics.py:100]: Changed

File: libcloudforensics/providers/azure/forensics.py [hunk 5]
```diff
@@ -123,10 +123,9 @@ def CreateDiskCopy(
     snapshot.Delete()
     logger.info('Disk {0:s} successfully copied to {1:s}'.format(
         disk_to_copy.name, new_disk.name))
-  except RuntimeError as exception:
-    error_msg = 'Cannot copy disk "{0:s}": {1!s}'.format(
-        str(disk_name), str(exception))
-    raise RuntimeError(error_msg)
+  except (errors.CFUError, RuntimeError) as exception:
+    raise RuntimeError('Cannot copy disk "{0:s}": {1!s}'.format(
```
[the remark this answers is not in the document]
Comment (giovannt0) [reply libcloudforensics/providers/azure/forensics.py:127]: Just an omission on my side, changed :)

File: libcloudforensics/errors.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Generic error wrapper"""
+
+from libcloudforensics import logging_utils
+
+
+class CFUError(Exception):
```
[the remark this answers is not in the document]
Comment (giovannt0) [reply libcloudforensics/errors.py:20]: Renamed to `LCFError`

File: libcloudforensics/errors.py [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+
+class ResourceNotFoundError(LCFError):
+  """Error when an issue with non-existent resource is encountered."""
+
+
+class ResourceCreationError(LCFError):
+  """Error when an issue with creating a new resource is encountered."""
+
+
```
Comment (tomchop) [inline libcloudforensics/errors.py:64]:
```suggestion

```

**Pull request merged**